torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1146,6 +1147,8 @@ def method_tests():
             NO_ARGS, 'symmetric_psd', (), NO_ARGS, [skipCPUIfNoLapack, skipCUDAIfNoMagma]),
         ('det', lambda: random_symmetric_pd_matrix(S),
             NO_ARGS, 'symmetric_pd', (), NO_ARGS, [skipCPUIfNoLapack, skipCUDAIfNoMagma]),
+        ('det', lambda: random_hermitian_matrix(S),
```

Comment:
this test is running for `torch.double` because:
1. `random_hermitian_matrix` returns a double tensor (which I think we should update to `torch.cdouble`) https://github.com/pytorch/pytorch/blob/298e0e0d573a90ed516f32051151c32e9b309c0f/torch/testing/_internal/common_utils.py#L1507
2. `det` doesn't currently exist in this list https://github.com/pytorch/pytorch/blob/298e0e0d573a90ed516f32051151c32e9b309c0f/tools/autograd/gen_variable_type.py#L156, so when backward is run with complex input, it would throw a `RuntimeError`.
3. To actually run a method_test for complex, you would have to have to add 'complex' to the substring so in this case, it would be `complex_hermitian' and also add the name of the test to this list: https://github.com/pytorch/pytorch/blob/298e0e0d573a90ed516f32051151c32e9b309c0f/test/test_autograd.py#L4825

Reply:
Thank you! Unfortunately, det_backward relies on svd_backward, and the latter does not yet work with complex input.